website/client/js/services/userServices.js

```diff
@@ -106,11 +106,11 @@ angular.module('habitrpg')
         })
         // refresh all but completed todos
         .then(function (response) {
-          tasks = response.data.data;
+          tasks.push.apply(tasks, response.data.data);
+
 
           // only refresh completed todos if the user has the completed tabs list open
-          var taskButton = document.getElementById('completedTasksButton');
-          if (taskButton && taskButton.classList.contains('active')) {
+          if ($rootScope.lists && $rootScope.pageTitle == 'Tasks' && _.find($rootScope.lists, {'type':'todo'}).view == 'complete') {
```

Comment:
Will asserting on the page title work if you're using a non-English locale? Probably better to use the $stare module.